crates/wasmparser/src/readers/core/types.rs

```diff
@@ -840,15 +1022,50 @@ pub struct TagType {
 }
 
 /// A reader for the type section of a WebAssembly module.
-pub type TypeSectionReader<'a> = SectionLimited<'a, Type>;
+pub type TypeSectionReader<'a> = SectionLimited<'a, SubType>;
 
-impl<'a> FromReader<'a> for Type {
+impl<'a> FromReader<'a> for StructuralType {
+    fn from_reader(reader: &mut BinaryReader<'a>) -> Result<Self> {
+        read_structural_type(reader.read_u8()?, reader)
+    }
+}
+
+fn read_structural_type(
+    op_code: u8,
```

Comment:
Nitpick: could we name this something other than `op_code` here and below in the `FromReader` impl for `SubType`? That makes me think of instruction opcodes where as this doesn't have anything to do with instructions, just types.

Perhaps `type_discriminant`? Or `leading_byte`?

Reply:
Would `opcode` (without the `_`) work? Seems to be the term used for this purpose in the spec (for example, in https://webassembly.github.io/spec/core/appendix/index-types.html)

Alternatively, how about `type_code`?

Reply:
Ah, I didn't realize the spec used `opcode` for this. In that case, yes, let's go with that!